libraries/botbuilder-ai-orchestrator/src/orchestratorAdaptiveRecognizer.ts

```diff
@@ -110,228 +90,196 @@ export class OrchestratorAdaptiveRecognizer extends Recognizer implements Orches
-            this.Initialize();
+        this._initializeModel();
+
+        if (this.externalEntityRecognizer) {
+            // Run external recognition
+            const externalResults = await this.externalEntityRecognizer.recognize(
+                dc,
+                activity,
+                telemetryProperties,
+                telemetryMetrics
+            );
+            recognizerResult.entities = externalResults.entities;
         }
 
-        // recognize text
-        const result = await this.resolver.score(text);
+        // Score with orchestrator
+        const results = await this._resolver.score(text);
 
-        if (Object.entries(result).length !== 0) {
-            this.AddTopScoringIntent(result, recognizerResult);
-        }
+        // Add full recognition result as a 'result' property
+        recognizerResult[this.resultProperty] = results;
 
-        // run entity recognizers
-        await this.recognizeEntities(dialogContext, text, activity.locale || '', recognizerResult);
+        if (results.length) {
+            const topScoringIntent = results[0].label.name;
+            const topScore = results[0].score;
 
-        // Add full recognition result as a 'result' property
-        recognizerResult[this.resultProperty] = result;
-
-        // disambiguate
-        if (detectAmbiguity) {
-            const topScoringIntent = result[0].score;
-            const scoreDelta = topScoringIntent - disambiguationScoreThreshold;
-            const ambiguousIntents = result.filter((x) => x.score >= scoreDelta);
-            if (ambiguousIntents.length > 1) {
-                recognizerResult.intents = {};
-                recognizerResult.intents[this.chooseIntent] = { score: 1.0 };
-                recognizerResult[this.candidatesCollection] = [];
-                ambiguousIntents.forEach((item) => {
-                    const itemRecoResult = {
-                        text: text,
-                        intents: {},
-                        entities: {},
-                        score: item.score,
-                    };
-                    itemRecoResult.intents[item.label.name] = {
-                        score: item.score,
-                    };
-                    itemRecoResult.entities = recognizerResult.entities;
-                    recognizerResult[this.candidatesCollection].push({
-                        intent: item.label.name,
-                        score: item.score,
-                        closestText: item.closest_text,
-                        result: itemRecoResult,
-                    });
-                });
+            // if top scoring intent is less than threshold, return None
+            if (topScore < this.unknownIntentFilterScore) {
+                recognizerResult.intents.None = { score: 1.0 };
+            } else {
+                // add top score
+                recognizerResult.intents[`${topScoringIntent}`] ??= {
```

Comment:
Why is the interpolation necessary?

Reply:
I got a lint warning here: 
![image](https://user-images.githubusercontent.com/946375/103071519-b542dd80-45fe-11eb-8023-de15e0cbb114.png)

Reply:
Ah - we really should tune or disable that lint rule. I believe it's intended to protect against using unsanitized user input for object keys, which isn't that concerning. Feel free to add `// eslint-disable-next-line security/detect-object-injection` to remove the lint warning.

Reply:
Submitted https://github.com/microsoft/botbuilder-js/issues/3161 to track.